GH-8705 Expose errorOnTimeout on MessagingGateway (#8712)
spring-projects/spring-integration: src/reference/antora/modules/ROOT/pages/gateway.adoc

However, if the timeout has been reached before the actual reply was produced, it could result in a 'null' return from the gateway method.
You should understand that the reply message (if produced) is sent to a reply channel after the gateway method invocation might have returned, so you must be aware of that and design your flow with it in mind.

> Also see an `errorOnTimeout` option to throw a `MessageTimeoutException` instead of returning `null`.

Reviewer: ```suggestion
Also see the `errorOnTimeout` property to throw a `MessageTimeoutException` instead of returning `null`, when a timeout occurs.
```

If the flow was completely synchronous, the reply is immediately available.
For asynchronous flows, the thread waits for up to this time.

> Starting with version 6.2, the `errorOnTimeout` option of an internal `MethodInvocationGateway` extension of the `MessagingGatewaySupport` is exposed to the `@MessagingGateway` and `GatewayEndpointSpec`.

Reviewer: ```suggestion
Starting with version 6.2, the `errorOnTimeout` property of the internal `MethodInvocationGateway` extension of the `MessagingGatewaySupport` is exposed on the `@MessagingGateway` and `GatewayEndpointSpec`.
```

This option has exactly the same meaning as for any inbound gateway explained in the end of xref:endpoint-summary.adoc#endpoint-summary[Endpoint Summary] chapter.

> In other words, setting this option to `true`, would lead to the `MessageTimeoutException` being thrown from a send-and-receive gateway operation instead of returning `null` when receive timeout is exhausted.

Reviewer: ```suggestion
In other words, setting this option to `true`, would lead to the `MessageTimeoutException` being thrown from a send-and-receive gateway operation instead of returning `null` when the receive timeout is exhausted.
```
